coordinator/src/node.rs

```diff
@@ -202,98 +203,205 @@ impl Node {
         };
 
-        Ok(())
+        let temporary_contract_id = self
+            .inner
+            .propose_dlc_channel(contract_input, trade_params.pubkey)
+            .await
+            .context("Could not propose DLC channel")?;
+
+        // After the DLC channel has been proposed the position can be created. This fixes
+        // https://github.com/get10101/10101/issues/537, where the position was created before the
+        // DLC was successfully proposed.
+        //
+        // Athough we can still run into inconsistencies (e.g. if `propose_dlc_channel` succeeds,
+        // but `persist_position_and_trade` doesn't), we are more likely to succeed with the new
+        // order.
+        //
+        // FIXME: We should not create a shadow representation (position) of the DLC struct, but
+        // rather imply the state from the DLC.
+        self.persist_position_and_trade(
+            conn,
+            trade_params,
+            temporary_contract_id,
+            leverage_coordinator,
+            stable,
+        )
     }
 
     async fn open_position(
         &self,
         conn: &mut PgConnection,
+        dlc_channel: SignedChannel,
         trade_params: &TradeParams,
-        coordinator_leverage: f32,
         stable: bool,
     ) -> Result<()> {
         let peer_id = trade_params.pubkey;
-        tracing::info!(%peer_id, ?trade_params, "Opening position");
+        let dlc_channel_id = dlc_channel.channel_id;
 
-        let margin_trader = margin_trader(trade_params);
-        let margin_coordinator = margin_coordinator(trade_params, coordinator_leverage);
+        tracing::info!(
+            %peer_id,
+            order_id = %trade_params.filled_with.order_id,
+            channel_id = %dlc_channel_id.to_hex(),
+            ?trade_params,
+            "Opening position"
+        );
+
+        let coordinator_collateral = dlc_channel.own_params.collateral;
+        let trader_collateral = dlc_channel.counter_params.collateral;
+        let total_collateral = coordinator_collateral + trader_collateral;
+
+        let initial_price = trade_params.filled_with.average_execution_price();
+
+        let leverage_coordinator = self.coordinator_leverage_for_trade(&trade_params.pubkey)?;
         let leverage_trader = trade_params.leverage;
-        let total_collateral = margin_coordinator + margin_trader;
 
-        let fee = order_matching_fee_taker(
+        let margin_coordinator = margin_coordinator(trade_params, leverage_coordinator);
+        let margin_trader = margin_trader(trade_params);
+
+        let order_matching_fee = order_matching_fee_taker(
             trade_params.quantity,
             trade_params.average_execution_price(),
         )
         .to_sat();
-        let initial_price = trade_params.filled_with.average_execution_price();
 
         let coordinator_direction = trade_params.direction.opposite();
 
+        let (updated_coordinator_collateral, coordinator_collateral_reserve) = {
```

Comment:
This is a very fair criticism.

How do you feel about using the word "collateral" for the money in the _channel_ and "margin" for the money being traded (thus, less than or equal to the collateral). Originally I thought this was a good idea, but I don't think it works because those two words probably mean the same thing?

Reply:
Maybe what I mentioned in my previous comment is orthogonal.

In any case, I've simplified this so that there is no need for this `updated_collateral` concept.